src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/pdf/PdfBoxService.java

```diff
@@ -0,0 +1,212 @@
+     * @param caseData contains the data needed to generate the PDF
+     * @param userToken contains the user authentication token
+     * @param caseTypeId reference for which casetype the document is being uploaded to
+     * @param documentName name of the document
+     * @param pdfTemplate location of the pdf template to be mapped with case data
+     * @return DocumentInfo which contains the URL and markup of the uploaded document
+     * @throws IOException if the call to Tornado has failed, an exception will be thrown. This could be due to
+    timeout or maybe a bad gateway.
+     */
+    public DocumentInfo generatePdfDocumentInfo(CaseData caseData, String userToken, String caseTypeId,
+                                                String documentName, String pdfTemplate)
+            throws IOException, PdfBoxServiceException {
+        if (ObjectUtils.isEmpty(caseData)) {
+            Throwable exception = new Throwable(PDF_SERVICE_EXCEPTION_FIRST_WORD_WHEN_CASE_DATA_EMPTY);
+            throw new PdfBoxServiceException(PDF_SERVICE_EXCEPTION_FIRST_WORD_WHEN_CASE_DATA_EMPTY, exception,
+                    PDF_SERVICE_EXCEPTION_FIRST_WORD_WHEN_CASE_DATA_EMPTY, CASE_DATA_NOT_FOUND_EXCEPTION_MESSAGE,
+                    PDF_SERVICE_CLASS_NAME, GENERATE_PDF_DOCUMENT_INFO_SERVICE_NAME);
+        }
+        throwPdfServiceExceptionWhenStringValueIsEmpty(userToken,
+                PDF_SERVICE_EXCEPTION_WHEN_USER_TOKEN_EMPTY,
+                PDF_SERVICE_EXCEPTION_WHEN_USER_TOKEN_EMPTY, caseData.getEthosCaseReference());
+        throwPdfServiceExceptionWhenStringValueIsEmpty(caseTypeId,
+                PDF_SERVICE_EXCEPTION_WHEN_CASE_TYPE_ID_EMPTY,
+                PDF_SERVICE_EXCEPTION_WHEN_CASE_TYPE_ID_EMPTY, caseData.getEthosCaseReference());
+        throwPdfServiceExceptionWhenStringValueIsEmpty(documentName,
+                PDF_SERVICE_EXCEPTION_WHEN_DOCUMENT_NAME_EMPTY,
+                PDF_SERVICE_EXCEPTION_WHEN_DOCUMENT_NAME_EMPTY, caseData.getEthosCaseReference());
+        throwPdfServiceExceptionWhenStringValueIsEmpty(pdfTemplate,
+                PDF_SERVICE_EXCEPTION_WHEN_PDF_TEMPLATE_EMPTY,
+                PDF_SERVICE_EXCEPTION_WHEN_PDF_TEMPLATE_EMPTY, caseData.getEthosCaseReference());
+        try {
+            byte[] bytes = convertCaseToPdfAsByteArray(caseData, pdfTemplate);
+            String dmStoreDocumentName = generatePdfFileName(caseData, documentName);
+            return createDocumentInfoFromBytes(userToken, bytes, dmStoreDocumentName, caseTypeId);
+        } catch (IOException ioe) {
+            logException(String.format(UNABLE_TO_PROCESS_PDF_SOURCE, pdfTemplate),
+                    caseData.getEthosCaseReference(), ioe.getMessage(), PDF_SERVICE_CLASS_NAME,
+                    GENERATE_PDF_DOCUMENT_INFO_SERVICE_NAME);
+            throw ioe;
+        }
+    }
+
+    private static void throwPdfServiceExceptionWhenStringValueIsEmpty(String stringValue,
```

Comment:
for all usages of this method, exceptionMessage is the same as firstWord. could this be refactored so firstWord or exceptionMessage is removed and we just pass it in once